### PR TITLE
fixed some grammar issue

### PR DESCRIPTION
**Pull request by punkdata**
The sentence did read clearly so I added a change to make it read a little cleaner

### CODE REVIEW

**Comment by punkdata** [comment]
this statement makes much more sense and prob should replace the other sentence

> Secrets engines operate by receiving API calls that meet this interface. The calls receive data from the caller, do some stuff, and return something.

reads alot cleaner

**Comment by tyrannosaurus-becks** [comment]
Thanks! I think it's not "formal" enough, though, for that doc. What if we did something like:

>Secrets engines operate by receiving API calls that meet [this interface](https://github.com/hashicorp/vault/blob/061bc8ed2bbed49eb4f54c3a04647fbbefb13e48/sdk/framework/backend.go#L105). The calls receive data from the caller, do something with that data, and return a result.

**Comment by punkdata** [comment]
LGTM

File: website/pages/docs/secrets/index.mdx [hunk 1]
```diff
@@ -9,7 +9,7 @@ description: Secrets engines are mountable engines that store or generate secret
 
 Secrets engines are components which store, generate, or encrypt data. Secrets
 engines are incredibly flexible, so it is easiest to think about them in terms
-of their function. Secrets engines are provided some set of data, they take some
+of their function. Secrets engines provide some set of data, they take some
```
Comment (tyrannosaurus-becks) [inline website/pages/docs/secrets/index.mdx:12]:
```suggestion
of their function. Secrets engines operate by receiving API calls that meet [this interface](https://github.com/hashicorp/vault/blob/061bc8ed2bbed49eb4f54c3a04647fbbefb13e48/sdk/framework/backend.go#L105). The calls receive data from the caller, take some
```

**Pull request merged**